pkg/openstackbaremetalset/baremetalhost.go

```diff
@@ -70,7 +70,11 @@ func BaremetalHostProvision(
 	// User data cloud-init secret
 	if userDataSecret == nil {
 		templateParameters := make(map[string]interface{})
-		templateParameters["AuthorizedKeys"] = strings.TrimSuffix(string(sshSecret.Data["authorized_keys"]), "\n")
+		// Prepare ssh_authorized_keys list for template
+		splitKeys := strings.Split(strings.TrimSuffix(string(sshSecret.Data["authorized_keys"]), "\n"), "\n")
+		sshKeys := make([]string, len(splitKeys))
+		sshKeys = append(sshKeys, splitKeys...)
+		templateParameters["AuthorizedKeys"] = sshKeys
```

Comment:
I think this can be cleaned up and simplified.  Right now it duplicates the total number of keys by injecting a blank one for every actual key.  The cloud-init ends up looking like this:

```
ssh_authorized_keys:
      - 
      - ssh-rsa somekeyyyyyyyyyyyyyyyyyyyyyyyy
```

I think it might be as simple as:

```
templateParameters["AuthorizedKeys"] = strings.Split(strings.TrimSuffix(string(sshSecret.Data["authorized_keys"]), "\n"), "\n")
```

Reply:
@abays What happens if the content the user provided have line breaks with more than one consecutive `\n`. I'd ensure that the keys passed to the template are not empty string to avoid such cases.

Reply:
Yeah, validation of some sort might be a good idea

Reply:
It depends on how the keys are in the secret data, whether the way it's expected by cloud-init (as a list) or how it's in authorized_keys file (multiple lines with newline separator).

Reply:
maybe we should be clear about the format then?

until we pushed a workaround (inject only one key), the framework was:
- loading the `~/.ssh/authorized_keys`
- generate the nodeset ConfigMap by injecting the file content as-is, in b64 format

We can push a proper list instead - so that the operator would just need to check if it's a string or a list; if string, it probably should enforce some format (either "only one key", or "use that separator only"?); if list, just nudge it in the cloud-init config file.

Any thoughts? Checking the input is more than probably mandatory at some point. And getting some proper testing around the generated cloud-init config file is also something we really should consider, in order to avoid future regressions/issues.

Any thoughts?

Reply:
This approach [1] works for OSPdO, where we assume that the `authorized_keys` in the `Secret` is a newline-separated string of keys.  i.e. we expect the format you'd see in a `~/.ssh/authorized_keys`.

[1] https://github.com/openstack-k8s-operators/osp-director-operator/pull/1043

Reply:
> where we assume that the authorized_keys in the Secret is a newline-separated string of keys
Should we just assume or document that and throw a proper error if it's not a string separated by newline char.

Reply:
@abays just updated this current patch, adding the check for non-empty string. Up to you if you want to import the patch from OSPdO instead of getting this one. I can also import the comment as-is to match your own commit - just lemme know :).